tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CheckAllTool/PackageCheckTool.cs

```diff
@@ -0,0 +1,204 @@
+                    return new CLICheckResponse(1, "", $"Package directory not found: {ex.Message}");
+                }
+                catch (NotSupportedException ex)
+                {
+                    SetFailure(1);
+                    return new CLICheckResponse(1, "", $"Unsupported language: {ex.Message}");
+                }
+                catch (Exception ex)
+                {
+                    SetFailure(1);
+                    logger.LogError(ex, "Failed to create language service");
+                    return new CLICheckResponse(1, "", $"Unable to determine language for package at: {packagePath}. Error: {ex.Message}");
+                }
+
+                return normalizedCheckName switch
+                {
+                    "all" => await RunAllChecks(packagePath, languageService, ct),
+                    "changelog" => await RunChangelogValidation(packagePath, languageService),
+                    "dependency" => await RunDependencyCheck(packagePath, languageService, ct),
+                    _ => new CLICheckResponse(1, "", $"Unknown check type: {checkName}. Valid options are: all, changelog, dependency")
+                };
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Unhandled exception while running package check");
+                SetFailure(1);
+                return new CLICheckResponse(1, ex.ToString(), $"Unhandled exception while running {checkName} check");
+            }
+        }
+
+        private async Task<CLICheckResponse> RunAllChecks(string packagePath, ILanguageRepoService languageService, CancellationToken ct)
+        {
+            logger.LogInformation("Running all validation checks");
+            
+            var results = new List<CLICheckResponse>();
+            var overallSuccess = true;
+
+            // Run dependency check
+            var dependencyCheckResult = await languageService.AnalyzeDependenciesAsync(packagePath, ct);
+            results.Add(dependencyCheckResult);
+            if (dependencyCheckResult.ExitCode != 0)
+            {
+                overallSuccess = false;
+            }
+
+            // Run changelog validation
+            var changelogValidationResult = await languageService.ValidateChangelogAsync(packagePath);
+            results.Add(changelogValidationResult);
+            if (changelogValidationResult.ExitCode != 0)
+            {
+                overallSuccess = false;
+            }
+
+            if (!overallSuccess) 
+            { 
+                SetFailure(1); 
+            }
+
+            var message = overallSuccess ? "All checks completed successfully" : "Some checks failed";
+            var combinedOutput = string.Join("\n", results.Select(r => r.CheckStatusDetails));
+            
+            return overallSuccess 
+                ? new CLICheckResponse(0, combinedOutput) 
+                : new CLICheckResponse(1, combinedOutput, message);
```

Comment:
The output formatter service does handle printing `IEnumerable<Response>` if you don't want to deal with managing the combined output (just return `List<CLICheckResponse>`).